tools/ci_build/github/apple/build_apple_framework.py

```diff
@@ -30,19 +31,17 @@ def _parse_build_settings(args):
 
     build_settings["build_osx_archs"] = build_settings_data.get("build_osx_archs", DEFAULT_BUILD_OSX_ARCHS)
 
-    build_params = []
     if "build_params" in build_settings_data:
-        build_params += build_settings_data["build_params"]
+        build_settings["build_params"] = build_settings_data.get("build_params")
```

Comment:
don't need to use dict.get() to handle a missing value as the containing `if` already ensures that "build_params" is present
```suggestion
        build_settings["build_params"] = build_settings_data["build_params"]
```